### PR TITLE
WB-180 Give modal titles unique ids

### PR DESCRIPTION
**Pull request by jandrade**
- Added unique ID to modal titles
- Created `<UniqueDialog />` component to either return the unique ID or return the custom ID already provided by the parent component.
- Modified both `ModalHeader` and `ModalDialog` to associate the the modal title with the Dialog using the `aria-labelledby` attribute.
- Added unit tests.

### CODE REVIEW

**Comment by jeresig** [comment]
Deploy preview for *wonder-blocks* ready!

Built with commit b9c506058eb08d0f6d6cf4503948860ccd6536c3

https://deploy-preview-396--wonder-blocks.netlify.com

File: packages/wonder-blocks-modal/components/unique-dialog.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+|};
+
+/**
+ * This is a Dialog wrapper that returns a unique Modal Title for A11Y purposes.
+ * This way, the wrapped Dialog will receive this custom ID and will use it to refer to a
+ * visible dialog title.
+ *
+ * @see https://www.w3.org/TR/wai-aria-practices/#dialog_roles_states_props
+ */
+export default class UniqueDialog extends React.Component<Props> {
```
Comment (jandrade) [inline packages/wonder-blocks-modal/components/unique-dialog.js]: Added this functionality in a new component for reusability/composition purposes. This way, if a dev needs to create his own Modal/Dialog variation (e.g. TwoPanelDialog), he can rely on this component to add this unique title to the dialog.

e.g.
```
<UniqueDialog>
    {(id) => <CustomTwoPaneDialog id={id}> .... </CustomTwoPaneDialog>}
</UniqueDialog>
```

Comment (somewhatabstract) [reply packages/wonder-blocks-modal/components/unique-dialog.js]: I wonder if we can make a more generic variation of this component. I mean, there's nothing `modal`-specific here except the scope. Perhaps we can elevate scope to props and then make this be a drop in replacement for `UniqueIDProvider`?

File: packages/wonder-blocks-modal/components/modal-dialog.js [hunk 2]
```diff
@@ -35,15 +35,22 @@ type Props = {|
      * Custom styles
      */
     style?: StyleType,
+
+    /**
+     * An optional id to provide a selector for the title element.
+     */
+    id?: string,
```
Comment (kevinbarabash) [inline packages/wonder-blocks-modal/components/modal-dialog.js]: Maybe change this to `"aria-labelledby"": string`.

File: packages/wonder-blocks-modal/components/modal-header.js [hunk 3]
```diff
@@ -32,6 +32,11 @@ type Props = {|
      * use when the item is used on a dark background.
      */
     light: boolean,
+
+    /**
+     * An optional id to provide a selector for the title element.
+     */
+    id?: string,
```
Comment (kevinbarabash) [inline packages/wonder-blocks-modal/components/modal-header.js]: Maybe change this to `titleId?: string`.  Should this be mandatory?

File: packages/wonder-blocks-modal/components/one-pane-dialog/one-pane-dialog.js [hunk 4]
```diff
@@ -80,6 +81,12 @@ type Props = {|
      * Test ID used for e2e testing.
      */
     testId?: string,
+
+    /**
+     * An optional id parameter for the title. If one is
+     * not provided, a unique id will be generated.
+     */
+    id?: string,
```
Comment (kevinbarabash) [inline packages/wonder-blocks-modal/components/one-pane-dialog/one-pane-dialog.js]: Maybe `"titleId"?: string`

File: packages/wonder-blocks-modal/components/modal-header.js (hunk 3 above)
Comment (jandrade) [reply packages/wonder-blocks-modal/components/modal-header.js]: Agreed, this should be mandatory if we want to enforce accessibility.

File: packages/wonder-blocks-modal/components/unique-dialog.js (hunk 1 above)
Comment (jandrade) [reply packages/wonder-blocks-modal/components/unique-dialog.js]: That makes total sense!. This task will be implemented here: https://khanacademy.atlassian.net/browse/WB-576

**Pull request merged**